lib/sinon/spy.js

```diff
@@ -210,6 +210,11 @@ var spyApi = {
             throw err;
         } catch (e) {/* empty */}
         push.call(this.errorsWithCallStack, err);
+        if (matchings) {
```

Comment:
As above, assume arrays from `matchingFakes` and prefer ES5.1 array extras over `for` loops.